nodes/node.go

```diff
@@ -41,10 +42,15 @@ type Node interface {
 	Init(*types.NodeConfig, ...NodeOption) error
 	Config() *types.NodeConfig
 	PreDeploy(configName, labCADir, labCARoot string) error
-	Deploy(context.Context, runtime.ContainerRuntime) error
-	PostDeploy(context.Context, runtime.ContainerRuntime, map[string]Node) error
+	Deploy(context.Context) error
+	PostDeploy(context.Context, map[string]Node) error
 	WithMgmtNet(*types.MgmtNet)
-	SaveConfig(context.Context, runtime.ContainerRuntime) error
+	WithRuntime(runtime.ContainerRuntime)
+	SaveConfig(context.Context) error
+	Delete(context.Context) error
+	GetImages() []string
+	GetName() string
+	GetRuntime() runtime.ContainerRuntime
```

Comment:
this is one of the things that allows nodes to override runtime. Nodes have runtime as a private field and `GetRuntime` is a getter function.

Reply:
You mean a node creates the runtime if overwritten and that's what gets added to the `c.Runtimes` map ?
This means potential multiple inits of runtimes

Wouldn't it be better to create all required runtimes before init of nodes, the set of configured runtimes can be retrieved from the `Topology` struct.
Then change `nodes.WithRuntime(runtime.ContainerRuntime)` to become `nodes.WithRuntimes(map[string]runtime.ContainerRuntime)`.
the node would then store whichever runtime it needs.

This would also allow `cvx` to use `docker` runtime if `ignite` is not installed, with a warning ? But not sure if we want that.

Reply:
The issue is that we'd need to let nodes `Init` in order to know if they decide to overwrite the runtime. I'll try to move the `rInit()` out of the node's `Init` function.

Reply:
@karimra I've changed this to `WithRuntime(globalRuntime string, allRuntimes map[string]runtime.ContainerRuntime)` and updated the PR. wdyt?

Reply:
this works as well, making it generic for the other nodes kind can be added in another PR I think.

Remains the issue of initializing the runtimes before the nodes, current cvx code creates an ignite client for each node.

By going through the topology struct, it's possible to get a `map[string]struct{}` holding the specified runtimes names under nodes/kinds/default as keys, adding the global one, initializing them as well as the global one and passing them as a map to the node init.

Reply:
I don't think runtimes are initialized multiple times now. Due to this 
```
	// By default, running in ignite runtime
	if igniteRuntime, ok := allRuntimes[runtime.IgniteRuntime]; ok {
		c.runtime = igniteRuntime
		return
	}
```

I get your point that doing this in a single place is easier to understand and I'd like to do this too but the problem I have is that runtime override is a node-specific option, i.e. some nodes may decide to override it and the logic is inside that node's package. By simply reading the topology file, you won't know what runtimes nodes would select, unless it's explicitly specificed in `nodeCfg.Runtime`.

Reply:
I see, I think we can add that in a future PR, 
getting a default runtime for a node kind without initializing the node.